sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -742,15 +751,49 @@ class Analyzer(
       case _ => plan
     }
 
+    private def tryResolveV2Relation(u: UnresolvedRelation): Option[DataSourceV2Relation] = {
```

Comment:
I think this should continue to use extractors instead of inline logic. The rules are cleaner when written with extractors. We might need to update them, but I'd like to avoid methods like this one that do basically the same thing.